### ISSUE TITLE
Login with fetch api from custom IViewState implementation

### ISSUE THREAD

**Issue by chdev77** [IdentityServer/IdentityServer3#3606]
I'm working on creating an aurelia app to replace the example of your custom IViewService implementation using angular.  Angular does a form post 'action'....and for mine I want to so something very similar but have form post call a method on my view model.  I need this for custom validation on the login form.

This is the core of my Sign In method for the form post.
```
postUrlEndcoded(url, requestBody){
    this.http.configure(config => {
        config.withBaseUrl(environment.urls.authority);
        config.withDefaults({
          headers: {
            "Content-Type": "application/x-www-form-urlencoded",
            "Access-Control-Allow-Origin": "http://localhost:9000/",
            "Access-Control-Allow-Origin": "http://localhost:13964/"
          }
        });
    });
    return new Promise((resolve) =>{
      this.http.fetch(url, {
        method: 'post',
        body: requestBody,
        credentials: 'include', //this might need changed to 'include' for cross domain cookies
        redirect: 'follow', //this follows the redirect to 'core/connect/authorize?'
        mode: 'cors'
      })
        .then(response => {
          debugger;
           if(response.status === 200)
              return resolve(response);
            else throw new Error(`Error in fetch post: ${response.json()}`)
        });
    });
  }
```
The issue is with **credentials:'include'**.  The POST works and authenticates the user, the GET for the redirect is causing a redirection issue. 

- Fetch API cannot load http://localhost:9000/application#id_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1…sQMyFIQCfgU7Do7H3_-bRyFv-B5dO7edrtR51bpP8.2b2fa4ca46180ded583124bf89f4191d. Response to preflight request doesn't pass access control check: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. Origin 'http://localhost:13964' is therefore not allowed access.

The IdSvr Cors policy as follows
`factory.CorsPolicyService = new Registration<ICorsPolicyService>(new InMemoryCorsPolicyService(Clients.Get()));`

The IdScr Client AllowedCorsOrigins setup correctly.  I'm not sure what else it could be.

**Comment by brockallen** [comment]
> IViewState 

I guess you mean the IViewService. But I don't really follow the rest of the question.

If you're trying to do ajax requests to the authorize endpoint, that won't work. It's designed to work with redirects in the main browser window (or a popup).

**Comment by chdev77** [comment]
I corrected it...yes the IViewService.

I'm not calling authorize directly. On my aurelia app outside of the IViewService and Identity server I have implemented oidc-client and aurelia-open-id-connect plugin.  The plugin assists with any routing or aurelia weirdness.  When I click on Login/SignUp it directs me to IdSvr and my custom IViewService views that are also in aurelia.  Here, instead of doing a form post action, I'm doing a form post to a method.  Inside this method is where the above code gets called.

The posts method calls out to something like "Request URL:http://localhost:13964/core/login?signin=e3898364762a2f1a45618de4acb43ce1", here it responds back with the authorize location header and imediatly calls GET to the authorize endpoint.  I'm not doing the GET call directly in my code...its doing it on its own.

![signinpost](https://cloud.githubusercontent.com/assets/7691322/24509357/c0febf66-152b-11e7-893a-d3bb4391df14.PNG)

![signin302](https://cloud.githubusercontent.com/assets/7691322/24509371/c9fc3d00-152b-11e7-84ae-f654745e8af2.PNG)

![signinredirect](https://cloud.githubusercontent.com/assets/7691322/24509386/d43afb94-152b-11e7-9001-df8e2d5f0c23.PNG)
